control-plane-operator/controllers/awsprivatelink/awsprivatelink_controller.go

```diff
@@ -299,51 +316,108 @@ func reconcileAWSEndpointService(ctx context.Context, awsEndpointService *hyperv
 			return fmt.Errorf("endpoint %s not found, resetting status", serviceName)
 		}
 		log.Info("endpoint exists", "endpointID", endpointID)
-		return nil
+		endpointDNSEntries = output.VpcEndpoints[0].DnsEntries
+	} else {
+		if hcp.Spec.Platform.Type != hyperv1.AWSPlatform || hcp.Spec.Platform.AWS == nil || hcp.Spec.Platform.AWS.CloudProviderConfig == nil {
+			return fmt.Errorf("AWS platform information not provided in HostedControlPlane")
+		}
+		input := &ec2.CreateVpcEndpointInput{
+			ServiceName:     aws.String(serviceName),
+			VpcId:           aws.String(hcp.Spec.Platform.AWS.CloudProviderConfig.VPC),
+			VpcEndpointType: aws.String(ec2.VpcEndpointTypeInterface),
+		}
+		if hcp.Spec.Platform.AWS.CloudProviderConfig.Subnet != nil && hcp.Spec.Platform.AWS.CloudProviderConfig.Subnet.ID != nil {
+			input.SubnetIds = []*string{hcp.Spec.Platform.AWS.CloudProviderConfig.Subnet.ID}
```

Comment:
would this work without setting `input.SubnetIds`? so should Subnet.ID be part of the hasAWSConfig check above?

Reply:
Technically it works.

If you don't specify the `SubnetIds` on the input to `CreateVpcEndpointWithContext`, the Endpoint will be created in the guest VPC, but no interfaces will be added on any of the subnets.  This also causes their to be no DNS entry associated with the endpoint and the registration of the `hypershift.local` CNAME records will be skipped, requiring manual intervention before nodes can join the guest cluster i.e. the Endpoint interfaces need to be added to the subnet(s) and DNS CNAMEs created.

Might be a better UX just to require it though.